### ISSUE TITLE
Not working with json-summary format

### ISSUE THREAD

**Issue by rwwagner90** [istanbuljs/nyc#815]
## Expected Behavior
Works with json-summary

## Observed Behavior
Does not work with json-summary

## Bonus Points! Code (or Repository) that Reproduces Issue

Hi! We have been discussing how to fail builds based on coverage thresholds in ember-cli-code-coverage, and we've been attempting to document how to use nyc to do so, however, it seems nyc does not like the json-summary format.

You can see our discussion here https://github.com/kategengler/ember-cli-code-coverage/pull/173

As far as we can tell, nyc is supposed to support all the formats, right? We thought perhaps there was a bug here or something we are missing with the configuration. Thanks in advance for your time and help!

**Comment by rwwagner90** [comment]
@bcoe @DanielRuf any ideas? I'm not sure if there is someone specific I should ping about this.

**Comment by DanielRuf** [comment]
Any specific reason why I was linked? I'm not part of the istanbuljs team afaik.

**Comment by rwwagner90** [comment]
@DanielRuf I just looked at the latest commits, trying to determine who to ping. If you know who else I should tag, let me know.

**Comment by DanielRuf** [comment]
You can check the blame (view) button on files if specific files are affected.
I just provided some CI related changes.

**Comment by rwwagner90** [comment]
I'm not sure where the issue lies. I just know that ember-cli-code-coverage was giving us issues when we tried to fail a build based on coverage with nyc because it did not like the json-summary format, which we require.

**Comment by DanielRuf** [comment]
A small example project and config to reproduce this could be very helpful.

Can you provide something for us so we can see the issue in live and let some tests run?

**Comment by rwwagner90** [comment]
@DanielRuf I linked the discussion when I reported this. kategengler/ember-cli-code-coverage#173

ember-cli-code-coverage reproduces the problem. I believe any json-summary formatted report will cause nyc to choke though. This comment does a good job of explaining the issue I think https://github.com/kategengler/ember-cli-code-coverage/pull/173#issuecomment-381980411

**Comment by rwwagner90** [comment]
Any updates on this?

**Comment by DanielRuf** [comment]
> Any updates on this?

Not sure who you mean, we are the only two people involved in this (participants).

**Comment by rwwagner90** [comment]
@DanielRuf I'm not sure how to get other people involved. Would be really awesome to start working toward a fix for this though.

**Comment by DanielRuf** [comment]
cc @istanbuljs

**Comment by rwwagner90** [comment]
It's not stale, still waiting on @istanbuljs

**Comment by JaKXz** [comment]
Sorry for the late reply, but can you clarify what you are looking for? Does the `json-summary` report not work? Is there something wrong with how the results are being processed and generated?

I'm not sure this is an `nyc` issue, but probably an underlying issue in https://github.com/istanbuljs/istanbuljs.

**Comment by rwwagner90** [comment]
@JaKXz no worries, the json-summary report does not work with nyc. You can see my initial comments and you can see the link to the issue where we discussed it further.
